app/file.go

```diff
@@ -360,7 +360,7 @@ func MigrateFilenamesToFileInfos(post *model.Post) []*model.FileInfo {
 
 func GeneratePublicLink(siteURL string, info *model.FileInfo) string {
 	hash := GeneratePublicLinkHash(info.Id, *utils.Cfg.FileSettings.PublicLinkSalt)
-	return fmt.Sprintf("%s%s/public/files/%v/get?h=%s", siteURL, model.API_URL_SUFFIX_V3, info.Id, hash)
+	return fmt.Sprintf("%s/files/%v/public?h=%s", siteURL, info.Id, hash)
```

Comment:
I think you'll actually need to make another function that's for v4 only, that way we can still use this one with v3 (hence the unit test failure)